assets/l10n/app_en.arb

```diff
@@ -27,18 +31,290 @@
+  "permissionsNeededError": "Error",
+  "@permissionsNeededError": {
+    "description": "General title for access dialog when we encounter an unknown error."
   },
-  "cameraAccessDeniedMessage": "To upload an image, please grant Zulip additional permissions in Settings.",
-  "@cameraAccessDeniedMessage": {
-    "description": "Message for dialog when the user needs to grant permissions for camera access."
+  "permissionsNeededTitle": "Permissions needed",
+  "@permissionsNeededTitle": {
+    "description": "Title for dialog when the user needs to grant additional permissions."
   },
-  "cameraAccessDeniedButtonText": "Open settings",
-  "@cameraAccessDeniedButtonText": {
+  "permissionsNeededOpenSettings": "Open settings",
+  "@permissionsNeededOpenSettings": {
+    "description": "Button label for permissions dialog button that opens the system settings screen."
+  },
+  "permissionsNeededCameraAccessDenied": "To upload an image, please grant Zulip additional permissions in Settings.",
+  "@permissionsNeededCameraAccessDenied": {
     "description": "Message for dialog when the user needs to grant permissions for camera access."
   },
+  "permissionsReadExternalStorageDenied": "To upload files, please grant Zulip additional permissions in Settings.",
+  "@permissionsReadExternalStorageDenied": {
+    "description": "Message for dialog when the user needs to grant permissions for external storage read access."
+  },
+  "actionSheetCopyMessageText": "Copy message text",
+  "@actionSheetCopyMessageText":  {
+    "description": "Label for copy message text button on action sheet."
+  },
+  "actionSheetCouldNotFetchMessageSource": "Could not fetch message source",
+  "@actionSheetCouldNotFetchMessageSource": {
+    "description": "Dialog message when the source of a message could not be fetched."
+  },
+  "actionSheetCopyingFailed": "Copying failed",
+  "@actionSheetCopyingFailed": {
+    "description": "Dialog message when copying the text of a message to the users system clipboard failed."
+  },
+  "actionSheetMessageCopied": "Message Copied",
+  "@actionSheetMessageCopied": {
+    "description": "Dialog message when content of a message was copied to the users system clipboard."
+  },
+  "actionSheetMessageDoesNotSeemToExist": "That message does not seem to exist.",
+  "@actionSheetMessageDoesNotSeemToExist": {
+    "description": "Dialog message in action sheet when loading a message that does not exist."
+  },
+  "actionSheetQuoteAndReply": "Quote and reply",
+  "@actionSheetQuoteAndReply":  {
+    "description": "Label for Quote and reply button on action sheet."
+  },
+  "actionSheetQuotationFailed": "Quotation failed",
+  "@actionSheetQuotationFailed": {
+    "description": "Error dialog message when quoting a message failed."
+  },
+  "actionSheetShare": "Share",
+  "@actionSheetShare":  {
+    "description": "Label for share button on action sheet."
```

Comment:
I like the "action sheet" prefix on these names. Perhaps a bit more explicitness to further scope what the context is about: `actionSheetOptionShare`.